core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -295,6 +295,10 @@ public MappedFieldType fieldType() {
         return fieldTypeRef.get();
     }
 
+    public FieldDataType fieldDataType() {
```

Comment:
Yes, the warming code is generic. It doesn't know right now what the concrete field mapper impl is.